cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceService.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.cloud.aws.connector.resource;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.ec2.model.DescribeInstanceTypesRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceTypesResult;
+import com.amazonaws.services.ec2.model.DiskInfo;
+import com.amazonaws.services.ec2.model.Filter;
+import com.amazonaws.services.ec2.model.InstanceStorageInfo;
+import com.google.api.client.util.Lists;
+import com.sequenceiq.cloudbreak.cloud.aws.AwsCloudFormationClient;
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonEc2Client;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AwsCredentialView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.instance.AwsInstaceStorageInfo;
+
+@Service
+public class AwsResourceService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsResourceService.class);
+
+    @Inject
+    private AwsCloudFormationClient awsClient;
+
+    public List<AwsInstaceStorageInfo> getInstanceTypeEphemeralInfo(AuthenticatedContext authenticatedContext, List<String> instanceTypes) {
```

Comment:
The functionality offered here - to figure out the ephemeral storage available on a specific instance type must already be available elsewhere in the code. The UI definitely knows how to configure this, and my guess is that backend components know this info to be able to mount disks / know how many disks to mount on a newly created cluster.

That should be used instead.

Reply:
There are different AWS endpoints with different responses. The only one which gives you if the ephemeral disk is attached to an instance is the one I am using. It is only used in one other place in the code to get recommendations for a cluster and it returns all the available instance types - which is a very big payload (makes it very very bulky to be used here). This is how the UI knows if an instance chosen has instance store. 

The other way the UI knows if the cluster has instance store is to look at the template table which says what type of storage is attached to a node. This doesn't go to AWS.

So i think I am using the right way to query this.

Reply:
I am referring to existing Cloudbreak code which performs the same operation. Is there an abstraction that is used by this bulky API that you're referring to ?